dlt/destinations/impl/databricks/databricks_adapter.py

```diff
@@ -9,34 +9,48 @@
 
 
 CLUSTER_HINT: Literal["x-databricks-cluster"] = "x-databricks-cluster"
+PARTITION_HINT: Literal["x-databricks-partition"] = "x-databricks-partition"
```

Comment:
there's already an "official" hint for column-based partitioning: `partition`. please use it.
FYI: we are working on unifying partition hint on the table level so the column order and transform functions (ie year(column)) can be defined